app/controllers/payments_controller.rb

```diff
@@ -0,0 +1,61 @@
+class PaymentsController < ApplicationController
+  before_action :set_payment, only: [:edit, :update, :destroy]
+  before_action :validate_first, only: [:new, :edit, :update, :create]
```

Comment:
revert back this method name change, and move these validations above set_payment.